analyses/cell-type-wilms-tumor-06/Dockerfile

```diff
@@ -0,0 +1,30 @@
+# pull base image
+FROM bioconductor/tidyverse:3.19
```

Comment:
I had a problem building this locally, but it seems that the base image might be the issue. I will need to investigate a little more.

Reply:
I changed for this:

```shell
# Base image on the Bioconductor 3.19 image
FROM bioconductor/r-ver:3.19
``` 

and build locally like this 
```shell
podman buildx build . -t openscpca/cell-type-wilms-tumor-06:latest --platform linux/amd64
``` 
and it works :)

Reply:
I am building locally to make some tested recommendations regarding what to do with `run.sh`, but it is taking a while, so I'll post an update as soon as possible!

Reply:
Thank you that would be great! 

I am initiating a renv environment from the current R Session to then simplify the Dockerfile using the renv.lock (https://openscpca.readthedocs.io/en/latest/ensuring-repro/docker/docker-images/)

renv_init() is taking a while... I'll continue tomorrow :)

Reply:
I had trouble accessing RStudio on the image I built locally. I'm attempting to isolate the problem... but that unfortunately requires rebuilding things 😅 ⏳

Reply:
Thank you for being on it :) I have troubles trying to implement the renv() environment and to change for the bioconductor/r-ver:3.19 based image... So far I can only open RStudio using the following Dockerfile (just a bit "cleaned" compared to the PR).

```shell
# Base image on the Bioconductor 3.19 image
FROM bioconductor/tidyverse:3.19

# Set global R options
RUN echo "options(repos = 'https://cloud.r-project.org')" > $(R --no-echo --no-save -e "cat(Sys.getenv('R_HOME'))")/etc/Rprofile.site
ENV RETICULATE_MINICONDA_ENABLED=FALSE

RUN R --no-echo --no-restore --no-save -e "install.packages('remotes')"

RUN R -e "devtools::install_github('enblacar/SCpubr')"
RUN R -e "remotes::install_github('satijalab/seurat', 'seurat5', quiet = TRUE)"  # this also install patchwork (and others)
RUN R -e "remotes::install_github('satijalab/azimuth', quiet = TRUE)"            # this also install SingleCellExperiment, DT (and others)
RUN R  -e "remotes::install_github('cancerbits/DElegate')"
RUN R  -e "install.packages('viridis')"
RUN R  -e "install.packages('ggplotify')"
RUN R -e "BiocManager::install('edgeR')"


# make sure all R related binaries are in PATH in case we want to call them directly
ENV PATH ${R_HOME}/bin:$PATH
```

Reply:
@jaclyn-taroni FYI, I removed from the run.sh et config file unnecessary volume and path to data that are specific to our group. I realized it prevent the execution of the docker image if not defined!

Just in case it can help.

Reply:
Last comment of the week ;) 

I added the renv.lock file from the RStudio session of docker image I am running on my machine.

I am now trying to use it to build the image as described here: (https://openscpca.readthedocs.io/en/latest/ensuring-repro/docker/docker-images/), but building of the image is full of ERROR because of BiocManager version not matching the bioconductor version.. Will take some time of fine tunning  ⏳

But I think it's worth trying to have the Docker image in this format, then it might be easier to share/reproduce? What do you think @jaclyn-taroni ?

Reply:
I agree that our end goal should be to use `renv` as part of the Docker build process, but I don't know if that's necessarily where we need to start. I think it's fine to start installing packages one-by-one in the Dockerfile, and then view `renv` as a feature enhancement!

I am specifically struggling with what base image should be used right now. 

I expect folks working on this project (including all of our staff at the Data Lab) to be on Macs with Apple Silicon. 

My understanding is that you want to be able to develop using the RStudio Server IDE from within the running container. This is often part of my workflow, and I expect many other project participants might have this use case! (That is to say, I think we are bumping into a problem that will come up again and again in the project...)

However, I'm not sure we can use `bioconductor/tidyverse:3.19` as the base image and have it play nicely with multiple architectures (e.g., if someone using an ARM machine wants to build and use it locally). That's because I don't think `rocker/tidyverse`, which I understand to be the base of `bioconductor/tidyverse`, supports ARM: https://github.com/rocker-org/rocker-versioned2/issues/830

So, we might instead want to use the `bioconductor_docker` as the base:

```
# pull base image
FROM bioconductor/bioconductor_docker:3.19
```

And until we implement `renv`, I expect we can install the Tidyverse packages the same way suggested in that issue ☝🏻 

```
# pull base image
FROM bioconductor/bioconductor_docker:3.19

# Adds tidyverse packages & devtools
RUN /rocker_scripts/install_tidyverse.sh
```

Then I _believe_ we'd have an image we can build and push to Elastic Container Registry using GitHub Actions that can also be built and used locally on ARM machines. This compatibility seems like a good goal to me.

I would have liked to test this all conclusively, but installing Azimuth is taking a very long time for me 😅 Appreciate your patience, @maud-p!